crates/proof-of-sql/src/base/database/column_repetition_operation.rs

```diff
@@ -0,0 +1,178 @@
+use super::{
+    slice_operation::{repeat_elementwise, repeat_slice},
+    Column, ColumnType,
+};
+use crate::base::scalar::Scalar;
+use bumpalo::Bump;
+use core::iter::Iterator;
+
+#[allow(dead_code)]
+pub trait RepetitionOp {
+    fn op<T: Clone>(column: &[T], n: usize) -> impl Iterator<Item = T>;
+
+    /// Run a column repetition operation on a `Column`.
+    fn column_op<'a, S>(column: &Column<'a, S>, alloc: &'a Bump, n: usize) -> Column<'a, S>
+    where
+        S: Scalar,
+    {
+        let len = n * column.len();
+        match column.column_type() {
+            ColumnType::Boolean => {
+                let mut iter = Self::op(column.as_boolean().expect("Column types should match"), n);
+                Column::Boolean(alloc.alloc_slice_fill_with(len, |_| {
+                    iter.next().expect("Iterator should have enough elements")
+                }) as &[_])
```

Comment:
If you want to:

This is showing up enough, where it might be worth an extension trait to add `alloc_slice_fill_iter_of_length`. (Would apply to the union code as well.
```suggestion
                Column::Boolean(alloc.alloc_slice_fill_iter_of_length(len, 
                    Self::op(column.as_boolean().expect("Column types should match"), n)
                ) as &[_])
```

Reply:
Sure. If we see another instance of this I will add it.